include/alpaka/mem/buf/cpu/Copy.hpp

```diff
@@ -181,12 +181,13 @@ namespace alpaka
                             {
                                 meta::ndLoopIncIdx(
                                     extentWithoutInnermost,
-                                    [&](vec::Vec<DimMin1, ExtentSize> const & idx)
+                                    [&] ALPAKA_FN_HOST_ACC (vec::Vec<DimMin1, ExtentSize> const & idx)
                                     {
-                                        std::memcpy(
-                                            reinterpret_cast<void *>(this->m_dstMemNative + (vec::cast<DstSize>(idx) * dstPitchBytesWithoutOutmost).foldrAll(std::plus<DstSize>())),
-                                            reinterpret_cast<void const *>(this->m_srcMemNative + (vec::cast<SrcSize>(idx) * srcPitchBytesWithoutOutmost).foldrAll(std::plus<SrcSize>())),
-                                            static_cast<std::size_t>(this->m_extentWidthBytes));
+                                        auto dst = reinterpret_cast<uint8_t *>(this->m_dstMemNative + (vec::cast<DstSize>(idx) * dstPitchBytesWithoutOutmost).foldrAll(std::plus<DstSize>()));
+                                        auto const src = reinterpret_cast<uint8_t const *>(this->m_srcMemNative + (vec::cast<SrcSize>(idx) * srcPitchBytesWithoutOutmost).foldrAll(std::plus<SrcSize>()));
+                                        size_t const numBytes = static_cast<std::size_t>(this->m_extentWidthBytes);
+                                        for(size_t b = 0u; b < numBytes; ++b)
+                                            dst[b] = src[b];
```

Comment:
We do not have any measurements, but I would expect this code to be much slower.

Reply:
Could you please add `{ }` around the loop body?

Reply:
> We do not have any measurements, but I would expect this code to be much slower.

I played in the past with https://github.com/raas/mbw where different methods were tested to copy memory. As I remember correctly it is mostly independent.

Another solution would be to declare the meta function as pure HOST function but then we can not use it on the device side.

Reply:
@tdd11235813 already opened an issue for the suffix qualifier https://github.com/ROCm-Developer-Tools/HIP/issues/659
I will wait if someone from the HIP dev team is answering soon before I will invest more time into this.